### PR TITLE
Rule improvement

### PR DESCRIPTION
**Pull request by konefalg**

### CODE REVIEW

File: whylog/config/rule.py [hunk 1]
```diff
@@ -125,9 +135,12 @@ def _create_constraints_list(cls, parser_ids_mapper, user_rule_intent):
 
     @classmethod
     def from_dao(cls, serialized_rule, parsers):
+        # TODO: restore serialized_rule["linkage"] when UserRuleIntent will support rule linkage
         return Rule(
-            [parsers[cause] for cause in serialized_rule["causes"]],
-            parsers[serialized_rule["effect"]], serialized_rule["constraints"]
+            [
```
Comment (ppolewicz) [inline whylog/config/rule.py]: this looks ugly. assign the generated list to a temporary variable?

File: whylog/config/rule.py [hunk 2]
```diff
@@ -125,9 +135,12 @@ def _create_constraints_list(cls, parser_ids_mapper, user_rule_intent):
...
+                parsers[cause] for cause in serialized_rule["causes"]
+            ], parsers[serialized_rule["effect"]], serialized_rule["constraints"],
+            serialized_rule.get("linkage", Rule.LINKAGE_AND)
```
Comment (ppolewicz) [inline whylog/config/rule.py:142]: I don't think we need to keep support for old style of rules where linkage was not specified.

File: whylog/config/rule.py (hunk 1 above)
Comment (konefalg) [inline whylog/config/rule.py]: I'm afraid I don't understand your explanation.

File: whylog/config/rule.py (hunk 2 above)
Comment (konefalg) [inline whylog/config/rule.py:142]: As you can see at the todo-comment above, it will be removed when UserRuleIntent will support different rule linkages.

File: whylog/config/rule.py (hunk 1 above)
Comment (kgromadzki) [inline whylog/config/rule.py]: It is simple: create temporary variable  for every Rule constructor param.
